src/preferences/PreferencesBase.js

```diff
@@ -173,6 +174,14 @@ define(function (require, exports, module) {
                 prefFile.read({}, function (err, text) {
                     if (err) {
                         if (createIfNew) {
```

Comment:
I think we should have "move to trash" for state.json only which is not generally intended to be user-editable. 
We may not want silly errors (like a missing comma) in the brackets.json/.brackets.json file causing it to be moved to trash. Being served the corrupted brackets.json file for editing would be a better solution as it is currently.